### ISSUE TITLE
deno lsp can not format files with emojis inside

### ISSUE THREAD

**Issue by starkpsmichael** [denoland/deno#8941]
#### Expected Behaviour

deno lsp should support and format files with emojis inside.

#### Actual Behaviour

```
Starting Deno language server...
  version: 1.6.3 (release, x86_64-unknown-linux-gnu)
Connected to "vscode" 1.52.1
language server initialized
Server ready.
thread 'main' panicked at 'assertion failed: self.is_char_boundary(n)', /rustc/7eac88abb2e57e752f3302f02be5f3ce3d7adfb4/library/alloc/src/string.rs:1555:29
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
[Error - 10:46:37 PM] Connection to server got closed. Server will not be restarted.
[Error - 10:46:37 PM] Request textDocument/formatting failed.
Error: Connection got disposed.
	at Object.dispose (/root/.vscode-server/extensions/denoland.vscode-deno-canary-0.0.6/client/node_modules/vscode-jsonrpc/lib/main.js:904:25)
	at Object.dispose (/root/.vscode-server/extensions/denoland.vscode-deno-canary-0.0.6/client/node_modules/vscode-languageclient/lib/client.js:74:35)
	at LanguageClient.handleConnectionClosed (/root/.vscode-server/extensions/denoland.vscode-deno-canary-0.0.6/client/node_modules/vscode-languageclient/lib/client.js:2309:42)
	at LanguageClient.handleConnectionClosed (/root/.vscode-server/extensions/denoland.vscode-deno-canary-0.0.6/client/node_modules/vscode-languageclient/lib/main.js:155:15)
	at closeHandler (/root/.vscode-server/extensions/denoland.vscode-deno-canary-0.0.6/client/node_modules/vscode-languageclient/lib/client.js:2296:18)
	at CallbackList.invoke (/root/.vscode-server/extensions/denoland.vscode-deno-canary-0.0.6/client/node_modules/vscode-jsonrpc/lib/events.js:62:39)
	at Emitter.fire (/root/.vscode-server/extensions/denoland.vscode-deno-canary-0.0.6/client/node_modules/vscode-jsonrpc/lib/events.js:121:36)
	at closeHandler (/root/.vscode-server/extensions/denoland.vscode-deno-canary-0.0.6/client/node_modules/vscode-jsonrpc/lib/main.js:240:26)
	at CallbackList.invoke (/root/.vscode-server/extensions/denoland.vscode-deno-canary-0.0.6/client/node_modules/vscode-jsonrpc/lib/events.js:62:39)
	at Emitter.fire (/root/.vscode-server/extensions/denoland.vscode-deno-canary-0.0.6/client/node_modules/vscode-jsonrpc/lib/events.js:121:36)
	at StreamMessageReader.fireClose (/root/.vscode-server/extensions/denoland.vscode-deno-canary-0.0.6/client/node_modules/vscode-jsonrpc/lib/messageReader.js:111:27)
	at Socket.<anonymous> (/root/.vscode-server/extensions/denoland.vscode-deno-canary-0.0.6/client/node_modules/vscode-jsonrpc/lib/messageReader.js:151:46)
	at Socket.emit (events.js:228:7)
	at Pipe.<anonymous> (net.js:664:12)
```

#### Steps to Reproduce
1. Create a file with `globalThis.console.log(`💩`)`
2. Try to format it with vscode-deno-canary

#### Using
- deno 1.6.3 (release, x86_64-unknown-linux-gnu)
- v8 8.8.294
- typescript 4.1.3
- vscode 1.52.1
- denoland.vscode-deno-canary

**Comment by kitsonk** [comment]
Duplicate of #8753

**Comment by starkpsmichael** [comment]
Ok, same usecase, related error? - but latest canary (db3a1d6).

```sh
RUST_BACKTRACE=full deno run --unstable test.ts
error: Unexpected character '💩' at file:///root/srv/localhost/test.ts:1:23
```

```sh
Starting Deno language server...
  version: 1.6.3+db3a1d6 (release, x86_64-unknown-linux-gnu)
Connected to "vscode" 1.52.1
language server initialized
Server ready.
thread 'tokio-runtime-worker' panicked at 'byte index 25 is not a char boundary; it is inside '💩' (bytes 23..27) of `globalThis.console.log(💩)`', /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/dprint-core-0.34.0/src/formatting/utils/string_utils.rs:55:21
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: JoinError::Panic(...)', cli/lsp/language_server.rs:706:6
[Error - 4:53:00 PM] Connection to server got closed. Server will not be restarted.
[Error - 4:53:00 PM] Request textDocument/formatting failed.
Error: Connection got disposed.
	at Object.dispose (/root/.vscode-server/extensions/denoland.vscode-deno-canary-0.0.6/client/node_modules/vscode-jsonrpc/lib/main.js:904:25)
	at Object.dispose (/root/.vscode-server/extensions/denoland.vscode-deno-canary-0.0.6/client/node_modules/vscode-languageclient/lib/client.js:74:35)
	at LanguageClient.handleConnectionClosed (/root/.vscode-server/extensions/denoland.vscode-deno-canary-0.0.6/client/node_modules/vscode-languageclient/lib/client.js:2309:42)
	at LanguageClient.handleConnectionClosed (/root/.vscode-server/extensions/denoland.vscode-deno-canary-0.0.6/client/node_modules/vscode-languageclient/lib/main.js:155:15)
	at closeHandler (/root/.vscode-server/extensions/denoland.vscode-deno-canary-0.0.6/client/node_modules/vscode-languageclient/lib/client.js:2296:18)
	at CallbackList.invoke (/root/.vscode-server/extensions/denoland.vscode-deno-canary-0.0.6/client/node_modules/vscode-jsonrpc/lib/events.js:62:39)
	at Emitter.fire (/root/.vscode-server/extensions/denoland.vscode-deno-canary-0.0.6/client/node_modules/vscode-jsonrpc/lib/events.js:121:36)
	at closeHandler (/root/.vscode-server/extensions/denoland.vscode-deno-canary-0.0.6/client/node_modules/vscode-jsonrpc/lib/main.js:240:26)
	at CallbackList.invoke (/root/.vscode-server/extensions/denoland.vscode-deno-canary-0.0.6/client/node_modules/vscode-jsonrpc/lib/events.js:62:39)
	at Emitter.fire (/root/.vscode-server/extensions/denoland.vscode-deno-canary-0.0.6/client/node_modules/vscode-jsonrpc/lib/events.js:121:36)
	at StreamMessageWriter.fireClose (/root/.vscode-server/extensions/denoland.vscode-deno-canary-0.0.6/client/node_modules/vscode-jsonrpc/lib/messageWriter.js:39:27)
	at Socket.<anonymous> (/root/.vscode-server/extensions/denoland.vscode-deno-canary-0.0.6/client/node_modules/vscode-jsonrpc/lib/messageWriter.js:58:46)
	at Socket.emit (events.js:223:5)
	at Pipe.<anonymous> (net.js:664:12)
```

**Comment by lucacasonato** [comment]
Will be fixed by #9151.

**Comment by kitsonk** [comment]
And still a duplicate of #8753 😄
